ui/src/main/resources/FileManagerCode/RecentFilesMacro.xml

```diff
@@ -308,7 +323,8 @@
   #set ($fileDoc = $xwiki.getDocument($result))
   #if ($fileDoc)
     #getFile($fileDoc $file)
-    * {{html}}#mimetypeimg($file.mediaType.toLowerCase() $file.name.toLowerCase()){{/html}} [[$file.name&gt;&gt;$result]] $services.localization.render('fileManager.recentFiles.creator', ["[[$xwiki.getUserName($file.creator, false)&gt;&gt;$file.creator]]"]) $services.localization.render('fileManager.recentFiles.creationDate', [$xwiki.formatDate($file.creationDate)])
+    #getFileFolders($fileDoc $paths)
+    * {{html}}#mimetypeimg($file.mediaType.toLowerCase() $file.name.toLowerCase()){{/html}} (%title="$folders"%)[[$file.name&gt;&gt;$result]] $services.localization.render('fileManager.recentFiles.creator', ["[[$xwiki.getUserName($file.creator, false)&gt;&gt;$file.creator]]"]) $services.localization.render('fileManager.recentFiles.creationDate', [$xwiki.formatDate($file.creationDate)])
```

Comment:
The wiki syntax ``(%title="$folders"%)`` is very fragile, as the folder names could contain special characters. The rule for me is: use HTML for UI and wiki syntax for content. This is more UI than content.